libindy_vdr/src/vdr/vdr.rs

```diff
@@ -0,0 +1,570 @@
+                                    //     let pool = self
+                                    //         .pools
+                                    //         .get(&did.namespace)
+                                    //         .ok_or(err_msg(VdrErrorKind::Resolver, "Unkown namespace"))
+                                    //         .unwrap();
+                                    //     fetch_legacy_endpoint_with_runner(
+                                    //         pool,
+                                    //         &did.id,
+                                    //         Box::new(|result| {
+                                    //             doc.endpoint = result.ok();
+                                    //             let diddoc = Some(doc.to_value().unwrap());
+                                    //             let md = if let Metadata::DidDocumentMetadata(md) =
+                                    //                 metadata
+                                    //             {
+                                    //                 Some(md)
+                                    //             } else {
+                                    //                 None
+                                    //             };
+
+                                    //             let result = ResolutionResult {
+                                    //                 did_resolution_metadata: None,
+                                    //                 did_document: diddoc,
+                                    //                 did_document_metadata: md,
+                                    //             };
+
+                                    //             callback(Ok(
+                                    //                 serde_json::to_string_pretty(&result).unwrap()
+                                    //             ))
+                                    //         }),
+                                    //     );
+                                } else {
+                                    // let diddoc = Some(doc.to_value().unwrap());
+                                    // let md = if let Metadata::DidDocumentMetadata(md) = metadata {
+                                    //     Some(md)
+                                    // } else {
+                                    //     None
+                                    // };
+
+                                    // let result = ResolutionResult {
+                                    //     did_resolution_metadata: None,
+                                    //     did_document: diddoc,
+                                    //     did_document_metadata: md,
+                                    // };
+                                    // if !doc.diddoc_content.is_none() {
+                                    //     callback(Ok(serde_json::to_string_pretty(&result).unwrap()))
+                                    // }
+                                }
```

Comment:
This would be an example.

I do `GET_NYM` Tx and if there's no `diddoc_content` present I need to do another `GET_ATTRIB` tx to see if there's a legacy endpoint attached to this nym. However, I can't get the lifetime requirements for the callbacks to match.

Reply:
Hm. The callback pattern shouldn't be necessary outside of the `ffi` module, I think you can just write it as an async function and avoid the boxing.

Reply:
Ah never mind, that comes from using a PoolRunner rather than a Pool.